lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java

```diff
@@ -632,6 +678,43 @@ public void skip(int n)
         };
     }
 
+    public ValueDecoder<int[]> getInt64TimestampNanosToLongTimestampWithTimeZoneDecoder(ParquetEncoding encoding)
+    {
+        ValueDecoder<long[]> delegate = getLongDecoder(encoding);
+        int precision = ((TimestampWithTimeZoneType) field.getType()).getPrecision();
+
+        return new ValueDecoder<>()
+        {
+            @Override
+            public void init(SimpleSliceInputStream input)
+            {
+                delegate.init(input);
+            }
+
+            @Override
+            public void read(int[] values, int offset, int length)
+            {
+                long[] buffer = new long[length];
+                delegate.read(buffer, 0, length);
+                // decoded values are epochNanos, convert to (packed epochMillisUtc, picosOfMilli)
```

Comment:
`decoded values are epochNanos, round to lower precision and convert to (packed epochMillisUtc, picosOfMilli)`